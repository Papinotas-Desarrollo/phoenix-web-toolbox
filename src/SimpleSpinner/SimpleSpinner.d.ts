export interface SimpleSpinnerProps {
  /**
   * Alternative color to use in the spinner, default is #F3F3F3
   */
  altColor?: string,
  /**
   * Main color to use in the spinner, default is #3498DB,
   */
  mainColor?: string;
  /**
   * Size radius of the spinner in px, default value is 25
   */
  size?: number;
}

declare const SimpleSpinner: React.ComponentType<SimpleSpinnerProps>;

export default SimpleSpinner;
