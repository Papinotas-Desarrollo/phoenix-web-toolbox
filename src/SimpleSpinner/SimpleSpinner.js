/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles({
  '@keyframes spin': {
    '0%': { transform: 'rotate(0deg)' },
    '100%': { transform: 'rotate(360deg)' }
  },
  wrapper: {
    paddingTop: '6px',
    textAlign: 'center',
    '& > div': {
      margin: 'auto',
      border: '4px solid',
      borderColor: props => props.altColor,
      borderTop: '4px solid',
      borderTopColor: props => props.mainColor,
      borderRight: '4px solid',
      borderRightColor: props => props.mainColor,
      borderRadius: '50%',
      width: props => props.size,
      height: props => props.size,
      animation: '$spin 1s linear infinite'
    }
  }
});

const SimpleSpinner = props => {
  const classes = useStyles(props);
  return (
    <div className={classes.wrapper}>
      <div />
    </div>
  );
};

SimpleSpinner.defaultProps = {
  altColor: '#F3F3F3',
  mainColor: '#3498DB',
  size: 25
};

SimpleSpinner.propTypes = {
  altColor: PropTypes.string,
  mainColor: PropTypes.string,
  size: PropTypes.number
};

export default SimpleSpinner;
