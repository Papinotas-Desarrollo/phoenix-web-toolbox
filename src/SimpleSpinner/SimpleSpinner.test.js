import React from 'react';
import { cleanup, render } from '@testing-library/react';

import SimpleSpinner from './SimpleSpinner';

describe('SimpleSpinner component', () => {
  let wrapper;
  let container;

  describe('with props', () => {
    beforeAll(() => {
      wrapper = render(<SimpleSpinner size={20} mainColor="#000000" altColor="red" />);
      ({ container } = wrapper);
    });
    afterAll(cleanup);

    it('renders correctly', () => {
      expect(container).toMatchSnapshot();
    });
  });
});
