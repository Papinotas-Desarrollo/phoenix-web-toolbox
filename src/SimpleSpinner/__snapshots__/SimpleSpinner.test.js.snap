// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleSpinner component with props renders correctly 1`] = `
<div>
  <div
    class="makeStyles-wrapper-2 makeStyles-wrapper-3"
  >
    <div />
  </div>
</div>
`;
